### ISSUE TITLE
'uv_fs_read' and 'uv_fs_write' file method signatures

### ISSUE THREAD

**Issue by coderkevin** [thlorenz/libuv-dox#11]
I was going through the learnuv project and looking at uv_fs_read.  The method signature here differs from the actual method signature in uv.h

libuv-dox
```C
int uv_fs_read(uv_loop_t* loop,
               uv_fs_t* req,
               uv_file file,
               void* buf,
               size_t length,
               int64_t offset,
               uv_fs_cb cb);
```

uv.h
```C
int uv_fs_read(uv_loop_t* loop,
               uv_fs_t* req,
               uv_file file,
               const uv_buf_t bufs[],
               unsigned int nbufs,
               int64_t offset,
               uv_fs_cb cb);
```

The same goes for uv_fs_write.  I'm new to libuv, so am I missing something?

**Comment by thlorenz** [comment]
Yeah it has gone out of date with libuv. I'm mentioning in the README which libuv version this is based on.
So we could do the work and update the docs or just deprecate this repo in favor of the official uv docs which have much improved since when I create this.

I'm thinking of deprecating docs only, but keeping all the examples for reference.

What do you think?

**Comment by coderkevin** [comment]
I think that sounds reasonable.  The libuv docs are very good now.  Also, since I moved the references in learnuv to point to the libuv docs, that dependency no longer exists.

**Comment by thlorenz** [comment]
We can discuss here how to communicate that documentation only is deprecated and point users to official docs while still pointing out useful examples.

Any ideas? Feel free to PR as well if you can think of a good way.

**Comment by coderkevin** [comment]
I would think about just removing the .md files and leave the examples directory. Then updating the README as such.  libuv-dox becomes a bit of a misnomer then, but a short explanation in the README could clear that up?  Either that or rename to libuv-examples?

What's the state of the examples?  I haven't looked at them all.  Do they all still run with the latest version of libuv?

**Comment by thlorenz** [comment]
> libuv-dox becomes a bit of a misnomer then

We can rename to uv-examples :)

> What's the state of the examples?

They worked fine when I created them (uv 0.11) something, so some will be incompat by now.
Also the build is a mess (via a Makefile) -- I was avoiding gyp still, but nowadays I'd setup the build via gyp, similar to how I did in learnuv.

Unfortunately I have zero bandwidth to even look into this, so we could leave them as is for now until people complain ;).
Unless you have more time and are interested in getting your hands dirty.

**Comment by coderkevin** [comment]
> We can rename to uv-examples :)

I think that would make the most sense.

> Unfortunately I have zero bandwidth to even look into this, so we could leave them as is for now until > people complain ;).
> Unless you have more time and are interested in getting your hands dirty.

Maybe start with renaming and then update them as time permits?  I'm sure I can make some time to help incrementally.
